src/Features/LanguageServer/Microsoft.CodeAnalysis.LanguageServer/HostWorkspace/LoadedProject.cs

```diff
@@ -108,69 +112,78 @@ public async ValueTask<(ImmutableArray<CommandLineReference>, OutputKind)> Updat
             _mostRecentFileInfo?.Documents,
             DocumentFileInfoComparer.Instance,
             document => _projectSystemProject.AddSourceFile(document.FilePath, folders: document.Folders),
-            document => _projectSystemProject.RemoveSourceFile(document.FilePath));
+            document => _projectSystemProject.RemoveSourceFile(document.FilePath),
+            "Project {0} now has {1} source file(s).");
 
-        var metadataReferences = _optionsProcessor.GetParsedCommandLineArguments().MetadataReferences.Distinct();
+        var metadataReferences = _optionsProcessor.GetParsedCommandLineArguments().MetadataReferences;
 
         UpdateProjectSystemProjectCollection(
             metadataReferences,
             _mostRecentMetadataReferences,
             EqualityComparer<CommandLineReference>.Default, // CommandLineReference already implements equality
             reference => _projectSystemProject.AddMetadataReference(reference.Reference, reference.Properties),
-            reference => _projectSystemProject.RemoveMetadataReference(reference.Reference, reference.Properties));
+            reference => _projectSystemProject.RemoveMetadataReference(reference.Reference, reference.Properties),
+            "Project {0} now has {1} reference(s).");
 
         // Now that we've updated it hold onto the old list of references so we can remove them if there's a later update
         _mostRecentMetadataReferences = metadataReferences;
 
-        var analyzerReferences = _optionsProcessor.GetParsedCommandLineArguments().AnalyzerReferences.Distinct();
+        var analyzerReferences = _optionsProcessor.GetParsedCommandLineArguments().AnalyzerReferences;
 
         UpdateProjectSystemProjectCollection(
             analyzerReferences,
             _mostRecentAnalyzerReferences,
             EqualityComparer<CommandLineAnalyzerReference>.Default, // CommandLineAnalyzerReference already implements equality
             reference => _projectSystemProject.AddAnalyzerReference(reference.FilePath),
-            reference => _projectSystemProject.RemoveAnalyzerReference(reference.FilePath));
+            reference => _projectSystemProject.RemoveAnalyzerReference(reference.FilePath),
+            "Project {0} now has {1} analyzer reference(s).");
 
         _mostRecentAnalyzerReferences = analyzerReferences;
 
         UpdateProjectSystemProjectCollection(
-            newProjectInfo.AdditionalDocuments.Distinct(DocumentFileInfoComparer.Instance), // TODO: figure out why we have duplicates
-            _mostRecentFileInfo?.AdditionalDocuments.Distinct(DocumentFileInfoComparer.Instance),
+            newProjectInfo.AdditionalDocuments,
```

Comment:
is the todo comment resolved by something else?

Reply:
It was resolved by me realizing that sometimes folks just have their MSBuild targets adding an additional file twice, so it's not an interesting thing to chase. :smile: